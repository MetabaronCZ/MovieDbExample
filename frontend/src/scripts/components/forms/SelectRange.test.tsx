import {
  fireEvent,
  getAllByRole,
  getByRole,
  getByText,
  queryByRole,
} from '@testing-library/dom';
import { describe, expect, it, jest } from '@jest/globals';

import { SelectRange } from './SelectRange';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';
import { render } from '@testing-library/react';
import { act } from 'react';

describe('components/forms/SelectRange', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <SelectRange
        label="LABEL"
        from={1}
        to={8}
        values={[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]}
        onFrom={() => null}
        onTo={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const label = getByText(container, 'LABEL');
    expect(label).toBeInTheDocument();

    const handles = getAllByRole(container, 'button');
    expect(handles.length).toEqual(2);
    expect(handles[0]).toHaveTextContent('1');
    expect(handles[1]).toHaveTextContent('8');

    // no options list opened
    const list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();
  });

  it('should not render when no values', async () => {
    const { container } = await waitForComponent(
      <SelectRange
        label="LABEL"
        from={1}
        to={8}
        values={[]}
        onFrom={() => null}
        onTo={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();
    expect(container.innerHTML).toEqual('');
  });

  it('should throw on invalid value', () => {
    // supress console.error
    jest.spyOn(console, 'error').mockImplementation(() => jest.fn());

    // "from" value is out of options
    const invalidFrom = (): void => {
      render(
        <SelectRange
          from={0}
          to={1}
          values={[1, 2]}
          onFrom={() => null}
          onTo={() => null}
        />,
        { wrapper: TestComponentWrapper },
      );
    };
    expect(invalidFrom).toThrow();

    // "to" value is out of options
    const invalidTo = (): void => {
      render(
        <SelectRange
          from={1}
          to={3}
          values={[1, 2]}
          onFrom={() => null}
          onTo={() => null}
        />,
        { wrapper: TestComponentWrapper },
      );
    };
    expect(invalidTo).toThrow();

    // "from" value is greater than "to" value
    const invalidRange = (): void => {
      render(
        <SelectRange
          from={2}
          to={1}
          values={[1, 2]}
          onFrom={() => null}
          onTo={() => null}
        />,
        { wrapper: TestComponentWrapper },
      );
    };
    expect(invalidRange).toThrow();

    // restore console.error supression
    jest.restoreAllMocks();
  });

  it('should render formatted values', async () => {
    const { container } = await waitForComponent(
      <SelectRange
        from={3}
        to={7}
        values={[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]}
        formatValue={(value) => `${10 * value}%`}
        onFrom={() => null}
        onTo={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const handles = getAllByRole(container, 'button');
    expect(handles.length).toEqual(2);
    expect(handles[0]).toHaveTextContent('30%');
    expect(handles[1]).toHaveTextContent('70%');
  });

  it('should be openable / closeable', async () => {
    const { container } = await waitForComponent(
      <SelectRange
        from={1}
        to={3}
        values={[0, 1, 2, 3, 4]}
        onFrom={() => null}
        onTo={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    const handles = getAllByRole(container, 'button');
    expect(handles.length).toEqual(2);

    let list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();

    // open "from" Select by handle click
    act(() => {
      fireEvent.click(handles[0]);
    });
    expect(container).toMatchSnapshot();

    list = queryByRole(container, 'list');
    expect(list).toBeInTheDocument();

    if (!list) {
      throw new Error('Could not find SelectRange list of items!');
    }
    let items = getAllByRole(list, 'button');
    expect(items.length).toEqual(5);
    expect(items[0]).toHaveTextContent('0');
    expect(items[0]).not.toHaveAttribute('disabled');
    expect(items[1]).toHaveTextContent('1');
    expect(items[1]).not.toHaveAttribute('disabled');
    expect(items[2]).toHaveTextContent('2');
    expect(items[2]).not.toHaveAttribute('disabled');
    expect(items[3]).toHaveTextContent('3');
    expect(items[3]).not.toHaveAttribute('disabled');
    expect(items[4]).toHaveTextContent('4');
    expect(items[4]).toHaveAttribute('disabled');

    // close Select by Escape key
    act(() => {
      fireEvent.keyUp(document, { key: 'Escape' });
    });
    list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();

    // open "to" Select by handle click
    act(() => {
      fireEvent.click(handles[1]);
    });
    expect(container).toMatchSnapshot();

    list = queryByRole(container, 'list');
    expect(list).toBeInTheDocument();

    if (!list) {
      throw new Error('Could not find SelectRange list of items!');
    }
    items = getAllByRole(list, 'button');
    expect(items.length).toEqual(5);
    expect(items[0]).toHaveTextContent('0');
    expect(items[0]).toHaveAttribute('disabled');
    expect(items[1]).toHaveTextContent('1');
    expect(items[1]).not.toHaveAttribute('disabled');
    expect(items[2]).toHaveTextContent('2');
    expect(items[2]).not.toHaveAttribute('disabled');
    expect(items[3]).toHaveTextContent('3');
    expect(items[3]).not.toHaveAttribute('disabled');
    expect(items[4]).toHaveTextContent('4');
    expect(items[4]).not.toHaveAttribute('disabled');

    // close Select by Escape key
    act(() => {
      fireEvent.keyUp(document, { key: 'Escape' });
    });
    list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();
  });

  it('should be disabled when needed', async () => {
    const { container } = await waitForComponent(
      <SelectRange
        from={1}
        to={3}
        values={[0, 1, 2, 3, 4]}
        disabled
        onFrom={() => null}
        onTo={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    const handles = getAllByRole(container, 'button');
    expect(handles.length).toEqual(2);

    let list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();

    // try open "from" Select by handle click
    act(() => {
      fireEvent.click(handles[0]);
    });
    list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();

    // try open "to" Select by handle click
    act(() => {
      fireEvent.click(handles[1]);
    });
    list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();
  });

  it('should select items', async () => {
    const onFrom = jest.fn();
    const onTo = jest.fn();

    const { container } = await waitForComponent(
      <SelectRange
        from={1}
        to={3}
        values={[0, 1, 2, 3, 4]}
        onFrom={onFrom}
        onTo={onTo}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(onFrom).not.toHaveBeenCalled();
    expect(onTo).not.toHaveBeenCalled();

    const handles = getAllByRole(container, 'button');
    expect(handles.length).toEqual(2);
    expect(handles[0]).toHaveTextContent('1');
    expect(handles[1]).toHaveTextContent('3');

    // no options list opened
    let list = queryByRole(container, 'list');
    expect(list).not.toBeInTheDocument();

    // open "from" Select by handle click
    act(() => {
      fireEvent.click(handles[0]);
    });
    list = getByRole(container, 'list');
    let items = getAllByRole(list, 'button');

    // click 3rd item
    act(() => {
      fireEvent.click(items[2]);
    });
    expect(onFrom).toHaveBeenCalledTimes(1);
    expect(onFrom).toHaveBeenCalledWith(2); // 3rd item value
    expect(onTo).not.toHaveBeenCalled();

    // Select is closed after selection
    expect(queryByRole(container, 'list')).toEqual(null);

    // open "to" Select by handle click
    act(() => {
      fireEvent.click(handles[1]);
    });
    list = getByRole(container, 'list');
    items = getAllByRole(list, 'button');

    // click 5th item
    act(() => {
      fireEvent.click(items[4]);
    });
    expect(onTo).toHaveBeenCalledTimes(1);
    expect(onTo).toHaveBeenCalledWith(4); // 5th item value
    expect(onFrom).toHaveBeenCalledTimes(1); // just one previous call

    // Select is closed after selection
    expect(queryByRole(container, 'list')).toEqual(null);
  });
});
