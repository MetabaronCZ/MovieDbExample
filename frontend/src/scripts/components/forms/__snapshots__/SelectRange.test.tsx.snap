// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/forms/SelectRange should be openable / closeable 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
}

.c4 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c4:disabled {
  cursor: not-allowed;
}

.c6 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
}

.c5:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c5:focus {
  outline: 1px dashed #002d56;
}

.c5:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

.c7 {
  font-size: 14px;
  font-weight: 400;
  text-align: left;
}

.c8 {
  position: absolute;
  top: calc(100% + 4px);
  left: 0;
  width: 240px;
  overflow: hidden;
  z-index: 1000;
  background: #fff;
  border: 1px solid #002d56;
  box-shadow: 1px 1px 4px #dae0e3;
}

.c9 {
  list-style-type: none;
  overflow-y: auto;
  max-height: 360px;
}

.c10 {
  border-bottom: 1px solid #dae0e3;
}

.c10:last-child {
  border-bottom: none;
}

.c11 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  width: 100%;
  padding: 4px 8px;
  background: #fff;
  white-space: nowrap;
  text-align: left;
}

.c11:hover {
  background-color: #f2f7fa;
}

.c11:focus {
  outline: 1px dashed #002d56;
  outline-offset: -1px;
}

.c11:disabled {
  color: #98c3d9;
}

.c3 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c2 {
  position: relative;
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
  justify-content: start;
}

.c12::after {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  content: '-';
  font-weight: 700;
}

<div>
  <div
    class="c0"
  >
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4 c5"
            id="select-10"
            title="Zavřít"
            type="button"
          >
            <span
              class="c6 c7"
            >
              1
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-up fa-lg "
              data-icon="angle-up"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="c8"
          >
            <ul
              class="c9"
            >
              <li
                class="c10"
              >
                <button
                  class="c4 c11"
                  type="button"
                >
                  0
                </button>
              </li>
              <li
                class="c10"
              >
                <button
                  class="c4 c11"
                  type="button"
                >
                  1
                </button>
              </li>
              <li
                class="c10"
              >
                <button
                  class="c4 c11"
                  type="button"
                >
                  2
                </button>
              </li>
              <li
                class="c10"
              >
                <button
                  class="c4 c11"
                  type="button"
                >
                  3
                </button>
              </li>
              <li
                class="c10"
              >
                <button
                  class="c4 c11"
                  disabled=""
                  type="button"
                >
                  4
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </fieldset>
    <div
      class="c12"
    />
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4 c5"
            id="select-11"
            title="Otevřít"
            type="button"
          >
            <span
              class="c6 c7"
            >
              3
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`components/forms/SelectRange should be openable / closeable 2`] = `
.c0 {
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
}

.c4 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c4:disabled {
  cursor: not-allowed;
}

.c6 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
}

.c5:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c5:focus {
  outline: 1px dashed #002d56;
}

.c5:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

.c7 {
  font-size: 14px;
  font-weight: 400;
  text-align: left;
}

.c9 {
  position: absolute;
  top: calc(100% + 4px);
  left: 0;
  width: 240px;
  overflow: hidden;
  z-index: 1000;
  background: #fff;
  border: 1px solid #002d56;
  box-shadow: 1px 1px 4px #dae0e3;
}

.c10 {
  list-style-type: none;
  overflow-y: auto;
  max-height: 360px;
}

.c11 {
  border-bottom: 1px solid #dae0e3;
}

.c11:last-child {
  border-bottom: none;
}

.c12 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  width: 100%;
  padding: 4px 8px;
  background: #fff;
  white-space: nowrap;
  text-align: left;
}

.c12:hover {
  background-color: #f2f7fa;
}

.c12:focus {
  outline: 1px dashed #002d56;
  outline-offset: -1px;
}

.c12:disabled {
  color: #98c3d9;
}

.c3 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c2 {
  position: relative;
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
  justify-content: start;
}

.c8::after {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  content: '-';
  font-weight: 700;
}

<div>
  <div
    class="c0"
  >
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4 c5"
            id="select-10"
            title="Otevřít"
            type="button"
          >
            <span
              class="c6 c7"
            >
              1
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
    <div
      class="c8"
    />
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4 c5"
            id="select-11"
            title="Zavřít"
            type="button"
          >
            <span
              class="c6 c7"
            >
              3
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-up fa-lg "
              data-icon="angle-up"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="c9"
          >
            <ul
              class="c10"
            >
              <li
                class="c11"
              >
                <button
                  class="c4 c12"
                  disabled=""
                  type="button"
                >
                  0
                </button>
              </li>
              <li
                class="c11"
              >
                <button
                  class="c4 c12"
                  type="button"
                >
                  1
                </button>
              </li>
              <li
                class="c11"
              >
                <button
                  class="c4 c12"
                  type="button"
                >
                  2
                </button>
              </li>
              <li
                class="c11"
              >
                <button
                  class="c4 c12"
                  type="button"
                >
                  3
                </button>
              </li>
              <li
                class="c11"
              >
                <button
                  class="c4 c12"
                  type="button"
                >
                  4
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`components/forms/SelectRange should not render when no values 1`] = `<div />`;

exports[`components/forms/SelectRange should render formatted values 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
}

.c4 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c4:disabled {
  cursor: not-allowed;
}

.c6 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
}

.c5:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c5:focus {
  outline: 1px dashed #002d56;
}

.c5:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

.c7 {
  font-size: 14px;
  font-weight: 400;
  text-align: left;
}

.c3 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c2 {
  position: relative;
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
  justify-content: start;
}

.c8::after {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  content: '-';
  font-weight: 700;
}

<div>
  <div
    class="c0"
  >
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4 c5"
            id="select-8"
            title="Otevřít"
            type="button"
          >
            <span
              class="c6 c7"
            >
              30%
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
    <div
      class="c8"
    />
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4 c5"
            id="select-9"
            title="Otevřít"
            type="button"
          >
            <span
              class="c6 c7"
            >
              70%
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`components/forms/SelectRange should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
}

.c5 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c5:disabled {
  cursor: not-allowed;
}

.c7 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c6 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
}

.c6:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c6:focus {
  outline: 1px dashed #002d56;
}

.c6:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

.c8 {
  font-size: 14px;
  font-weight: 400;
  text-align: left;
}

.c4 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c3 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: flex;
  flex-direction: row;
  gap: 4px;
  white-space: nowrap;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c2 {
  position: relative;
  display: flex;
  flex-direction: row;
  gap: 8px;
  align-items: center;
  justify-content: start;
}

.c9::after {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  content: '-';
  font-weight: 700;
}

<div>
  <div
    class="c0"
  >
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <label
          class="c3 "
          for="select-0"
        >
          LABEL
        </label>
        <div
          class="c4"
        >
          <button
            class="c5 c6"
            id="select-0"
            title="Otevřít"
            type="button"
          >
            <span
              class="c7 c8"
            >
              1
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
    <div
      class="c9"
    />
    <fieldset
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c4"
        >
          <button
            class="c5 c6"
            id="select-1"
            title="Otevřít"
            type="button"
          >
            <span
              class="c7 c8"
            >
              8
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down fa-lg "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
